### DISCUSSION TITLE
module-info.java 2.11.0 gives error on requires com.google.gson;

### DISCUSSION THREAD

**Discussion by AixNPanes** [google/gson#2721] [Q&A] [answered]
I tried to upgrade Gson from 2.10.1 to 2.11.0. My module-info.java now gives an error on "requies com.google.gson;" in the eclipse 2024-03 editor for module-info.java. maven 3.9.8 runs OK.

eclipse: 2024-03
maven: 3.9.8
JDK21: Zulu21.36+17-CA

pom.xml:
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
	<modelVersion>4.0.0</modelVersion>
	<parent>
		<groupId>org.springframework.boot</groupId>
		<artifactId>spring-boot-starter-parent</artifactId>
		<version>3.4.0-M1</version>
		<relativePath/> 
	</parent>
	<groupId>org-test</groupId>
	<artifactId>test</artifactId>
	<version>0.1.81</version>
	<name>test</name>
	<description>test</description>

	<properties>
		<start-class>org.test.Test</start-class>
		<spring-boot.version>3.4.0-M1</spring-boot.version>
		<java.version>21</java.version>
	</properties>

	<dependencies>
		<dependency>
			<groupId>com.google.code.gson</groupId>
			<artifactId>gson</artifactId>
			<version>2.11.0</version>
		</dependency>
	</dependencies>
	<repositories>
		<repository>
			<id>spring-milestones</id>
			<name>Spring Milestones</name>
			<url>https://repo.spring.io/milestone</url>
		</repository>
		<repository>
			<id>central</id>
			<name>Maven Central</name>
			<url>https://repo1.maven.org/maven2/</url>
		</repository>
	</repositories>
</project>

module-info.java:
open module ws.daley.pihome.environ
{
	exports org.test;

	requires com.google.gson;
}

org.test.Test.java:
package org.test;

public class Test
{
	public static void main(String[] strings) {
	}
}

Note that gson 20.10.1 has a module-info.class in the root directory of the jar. In gson 20.11.1 it has been moved to META-INF/versions/9

**Comment by AixNPanes** [comment]
make that 2.10.1 and 1.11.0

**Reply by Marcono1234** [reply]
You can edit your post here by clicking the "..." and selecting "Edit":
![Button screenshot](https://github.com/user-attachments/assets/172c5168-7e25-4f54-b0d7-db69af8d36e0)
(highlighted yellow here)

**Comment by Marcono1234** [comment, marked as answer]
> Note that gson 20.10.1 has a module-info.class in the root directory of the jar. In gson 20.11.1 it has been moved to META-INF/versions/9

Gson 2.10.1 had the `module-info.class` under `META-INF/versions/9` as well.

We had a report about a similar issue before: #2679
As mentioned there maybe the Gson JAR changed in some way when we updated Maven plugins or made the build reproducible, however the JAR should still be valid. It seems this might be an issue with Eclipse (see https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2495) since as you mention this only affects Eclipse but does not occur when running Maven from the command line.

**Comment by AixNPanes** [comment]
The problem occurred with spring-boot 3.4.0-M1 introducing Gson 2.11.0 onto an eclipse that doesn't handle multi-release jars (see https://github.com/eclipse-jdt/eclipse.jdt.core/issues/2495). Also, in my 2.10.1 jar there is no module-info.class anywhere in META-INF only in the root of the jar.

Anyway, I think this can be closed as a problem with eclipse.

**Reply by Marcono1234** [reply]
> Anyway, I think this can be closed as a problem with eclipse.

Ok, I have marked my comment above as answer. I hope that is ok for you.

**Comment by AixNPanes** [comment]
That is fine! Thanks!
